File: backend/internal/model/preset.go
package model

import "time"

type Preset struct {
	ID           string    `json:"id"`
	UserID       string    `json:"user_id"`
	Label        string    `json:"label"`
	Color        string    `json:"color"`
	DisplayOrder int       `json:"display_order"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type CreatePresetRequest struct {
	Label        string `json:"label"`
	Color        string `json:"color"`
	DisplayOrder int    `json:"display_order"`
}

type UpdatePresetRequest struct {
	Label        string `json:"label"`
	Color        string `json:"color"`
	DisplayOrder int    `json:"display_order"`
}

type PresetResponse struct {
	ID           string `json:"id"`
	Label        string `json:"label"`
	Color        string `json:"color"`
	DisplayOrder int    `json:"display_order"`
}

func (p *Preset) ToResponse() PresetResponse {
	return PresetResponse{
		ID:           p.ID,
		Label:        p.Label,
		Color:        p.Color,
		DisplayOrder: p.DisplayOrder,
	}
}

File: frontend/lib/api/preset.ts
import {
  CreatePresetRequest,
  PresetResponse,
  UpdatePresetRequest,
} from "@/types/preset";
import { API_URL, authHeaders, unwrap } from ".";

export const getUserPresets = async (token: string) => {
  const res = await fetch(`${API_URL}/presets`, {
    method: "GET",
    headers: authHeaders(token),
  });
  return unwrap<PresetResponse[]>(res);
};

export const createPreset = async (token: string, req: CreatePresetRequest) => {
  const res = await fetch(`${API_URL}/presets`, {
    method: "POST",
    headers: authHeaders(token),
    body: JSON.stringify({
      label: req.label,
      color: req.color,
      display_order: req.displayOrder,
    }),
  });
  return unwrap<PresetResponse>(res);
};

export const updatePreset = async (
  token: string,
  id: string,
  req: UpdatePresetRequest,
) => {
  const res = await fetch(`${API_URL}/presets/${id}`, {
    method: "PATCH",
    headers: authHeaders(token),
    body: JSON.stringify({
      label: req.label,
      color: req.color,
      display_order: req.displayOrder,
    }),
  });
  return unwrap<PresetResponse>(res);
};

export const deletePreset = async (token: string, id: string) => {
  const res = await fetch(`${API_URL}/presets/${id}`, {
    method: "DELETE",
    headers: authHeaders(token),
  });
  if (!res.ok) throw new Error(`HTTP error: ${res.status}`);
};
